src/Microsoft.Data/StringColumn.BinaryOperations.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.Data
+{
+    public partial class StringColumn : BaseColumn
+    {
+        public override BaseColumn Add(BaseColumn column)
```

Comment:
Do we want strongly typed versions of these methods? That way if I have a `StringColumn`, and I call one of these methods, I should get back a `StringColumn`, not?

Reply:
Hmmm, so at the moment, I've taken the less strongly typed approach and since one of the main use cases is notebooks, it pairs well with returning a copy everytime(since I don't know the return type until runtime). In the case of a `StringColumn`, `Add` will always returns a `StringColumn`, but for `intColumn + 5.0`, the result will be a `PrimitiveColumn<double>` for example. To enable the strong typed versions, I would have to define APIs for all possible combinations of all columns. How about I consider that in a separate issue? Since these methods are general, it's easy enough to build the strong APIs on top of this?

Reply:
Yes, a separate issue for this sounds good to me.

As for every combination of type, I don't think I'd go that far.  But for the case where I have a strongly-typed `string` or `Int32` column, if I add a `string` or `Int32` to that column, I know I get back a `string` or `Int32`.

The analogy here is this code:

```C#
string a = "foo";
string b = "bar";

??? c = a + b;
````

It's pretty obvious that `c` should be a `string`, not `object`.

Reply:
Filed [here](https://github.com/dotnet/corefxlab/issues/2684)